### ISSUE TITLE
Arreglos y Cadenas

### ISSUE THREAD

**Issue by DamRCorba** [DamRCorba/Informatica2_2_11#1]
En este apartado podrán hacer las consultas relacionadas a Arreglos y Cadenas.

**Comment by FrancoPerta** [comment]
Los lunes vamos a tener clases virtuales por zoom?? o solo tenemos que ver y hacer lo que se sube en GitHub??

**Comment by DamRCorba** [comment]
Hola Franco, dependiendo del tema haremos clases sincronicas con google meets, o asincronicas con esta metodologia para que ustedes resuelvan y practiquen. A programar se aprende programando. Los lunes corresponden a la practica. Es por eso que en documento de la clase 1 tienen ejercicios, ejemplos y videos. Ademas de este canal de consulta, el cual esta pensado para que lo usens en cualquier momento y no unicamente durante el tiempo de clase.

**Comment by peressantiago** [comment]
pero hoy tendremos clases via Google meet?

**Comment by DamRCorba** [comment]
No, como les dije en el correo, la clase de hoy es asincronica.

**Comment by peressantiago** [comment]
Por mi parte no me llego el correo de que la clase de hoy es asincrónica.  Necesita que le pase mi correo?

**Comment by DamRCorba** [comment]
Por favor escribime a mi correo para verificar tu mail, fijate por las dudas en spam. Igualmente, el correo que pase tenia el link de los datos de git [click aqui para Clase01](https://github.com/DamRCorba/Informatica2_2_11/tree/master/Clase01), y el link para las consultas que son esta pagina.

**Comment by peressantiago** [comment]
ahí lo encontré, estaba en el spam., mil disculpas por las molestias.

**Comment by LisandroGanduglia** [comment]
Buenas profe, puse el ejemplo 1 (calculo de seno con vectores) en el compilador y al ejecutar el programa para todos los angulos me da 0. No se si sera por el compilador, estoy usando el Dev c++. Gracias

**Comment by DamRCorba** [comment]
Muy bien Lisandro, no es un error del compilador. Hay  una variable cuyo tipo no es correcto.

**Comment by Peraltlu** [comment]
Profe, el compilador lo trabaja en radianes. ¿Es posible que lo trabaje en grados sexagesimales directo desde la biblioteca o eso debe hacerlo el programador? ¿Dónde puedo ver en qué sistema trabaja la bibliotecta? Yo uso también el Dev c++.

**Comment by DamRCorba** [comment]
Muy bien. La funcion en math.h, la cual es parte del ansi C y no depende del compilador, tiene el siguiente prototipo
```c
double sin(double ran);
```
Donde como parámetro espera un radian y no un angulo polar. 

Para hacerlo con angulos polares tenes que hacer la conversion. Por ejemplo:
```c
#include <stdio.h>
#include <math.h>

#define PI 3.14159265

int main () {
   double x, ret, val;

   x = 45.0;
   val = PI / 180;
   ret = sin(x*val);
   printf("El seno de %lf es %lf ", x, ret);
   
   return(0);
}
```
Que te daria una salida 
 > El seno de 45.00000 es 0.707107

La libreria ***<math.h>*** es parte de las librerias base de ANSI C con la que trabajan los compiladores. Puede haber diferencias por el sistema operativo. Si usas dev++ o codeblocks estas usando una version estable no muy actual del compilador, pero para la materia no vas a tener problemas o incluso va a ser mejor.
Si vas a un sistema operativo Linux, todas las distro usan la ultima version de GCC por lo que hay librerias que no vas a tener, ya que fueron eliminadas por cuestiones de seguridad. En las ultimas versiones de GCC no tenes las funciones ***gets*** y ***puts*** por ejemplo.
